(fix): HybridExecutor tasks of other executor rescheduled in kubernetes executor

pavansharma36 opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

Closes: #42151 


<!-- Please keep an empty line above the dashes. -->
---

pavansharma36 committed:
Fixed tasks of other executor rescheduled in kubernetes executor

pavansharma36 committed:
Fixed tasks of other executor rescheduled in kubernetes executor

pavansharma36 committed:
removed query logging

uranusjr commented on airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py:
```diff
@@ -197,13 +199,25 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
+                )
+            else:
+                criteria.append(TaskInstance.executor == KUBERNETES_EXECUTOR)
+            query = select(TaskInstance).where(*criteria)
```
I’d just do

```python
query = select(TaskInstance).where(
    TaskInstance.state == TaskInstanceState.QUEUED,
    TaskInstance.queued_by_job_id == self.job_id,
)
if self.kubernetes_queue:
    query = query.where(TaskInstance.queue == self.kubernetes_queue)
if KUBERNETES_EXECUTOR == default_executor:
    query = query.where(
        or_(
            TaskInstance.executor == KUBERNETES_EXECUTOR,
            TaskInstance.executor.is_(None),
        ),
    )
else:
    query = query.where(TaskInstance.executor == KUBERNETES_EXECUTOR)
```

Easier to read IMO.

pavansharma36 replied:
@uranusjr  done.

PS kubernetes executor check added in elif block as if queue name is provided then no need to check for executors

pavansharma36 committed:
Merge branch 'apache:main' into main

pavansharma36 committed:
review comments fix

pavansharma36 committed:
backward compaitibility

pavansharma36 committed:
backward compaitibility

pavansharma36 committed:
Merge branch 'main' into main

pavansharma36 committed:
Merge branch 'main' into main

o-nikolas commented on airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py:
```diff
@@ -199,13 +200,32 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
             )
             if self.kubernetes_queue:
                 query = query.where(TaskInstance.queue == self.kubernetes_queue)
+            elif has_executor and KUBERNETES_EXECUTOR == default_executor:
```
Shouldn't this just be an if? It's not going to get evaluated if `self.kubernetes_queue` exists, and those don't feel mutually exclusive to me?

pavansharma36 replied:
This exclusive for celery kubernetes executor (deprecated) so if its that case no need to check for hybrid executor

o-nikolas commented on airflow/providers/cncf/kubernetes/executors/kubernetes_executor.py:
```diff
@@ -199,13 +200,32 @@ def clear_not_launched_queued_tasks(self, session: Session = NEW_SESSION) -> Non
 
+        has_executor = hasattr(TaskInstance, "executor")
+        default_executor = None
+        if has_executor:
```
`has_executor` is being used here as a mechanism to check if multiple executor configuration is available right? For backcompat?

If so, I'd rename this variable because it makes the code hard to grok if you don't  have the details and there are no comments.

Maybe `hybrid_executor_enabled`?

o-nikolas commented on tests/providers/cncf/kubernetes/executors/test_kubernetes_executor.py:
```diff
@@ -1247,6 +1250,11 @@ def test_clear_not_launched_queued_tasks_not_launched(
         mock_kube_dynamic_client.return_value.resources.get.return_value = mock_pod_resource
         mock_kube_dynamic_client.return_value.get.return_value.items = []
 
+        # This is hack to use overridden conf vars as it seems executors loaded before conf override.
```
The ExecutorLoader holds state at the module level so that it's only loaded once and unique across instances. But that leads to weird stuff like this in tests. If you look at other examples we use strategic calls to `reload()` that module. I haven't verified if what you're doing here works just as well.

pavansharma36 replied:
yes clear works, but reload seems better, updated

o-nikolas commented on tests/providers/cncf/kubernetes/executors/test_kubernetes_executor.py:
```diff
@@ -1419,7 +1445,128 @@ def test_clear_not_launched_queued_tasks_not_launched_other_queue(
+        executor.clear_not_launched_queued_tasks(session=session)
+
+        ti.refresh_from_db()
+        assert ti.executor == "CeleryExecutor"
```
Can you also add a test case for when there is default executor that is not k8s and we should expect no change to state as well.

pavansharma36 replied:
I think test_clear_not_launched_queued_tasks_not_launched_other_executor and test_clear_not_launched_queued_tasks_launched_none_executor covers that

o-nikolas replied:
Both those cases have k8s as the default executor though? I don't see how those cover the case where k8s is _not_ the default executor.

pavansharma36 replied:
added

pavansharma36 committed:
resolve comments

pavansharma36 committed:
Merge branch 'main' into main

pavansharma36 committed:
added other default executor test

pavansharma36 committed:
Merge branch 'main' into main

pavansharma36 committed:
Merge branch 'main' into main

gopidesupavan committed:
Updating prometheus statsd exporter version from v0.27.2-pr579 to v0.27.2 and pre-commit-hooks to v5.0.0 (#42809)

dstandish committed:
Specify asyncio_default_fixture_loop_scope in pytest.ini (#42827)

Previously we were getting 'PytestDeprecationWarning: The configuration option "asyncio_default_fixture_loop_scope" is unset.'  This sets it to the new default, and makes the warning go away.

EvertonSA committed:
feat: allow customizing podManagementPolicy (#42673)

* feat: allow customizing podManagementPolicy

* fix: adjust schema and identation

* ci: adjust lint

---------

Co-authored-by: Arakaki, Everton <everton.arakaki@rabobank.nl>

dstandish committed:
Fix import sequencing of backfill and dagrun models (#42828)

Previously it worked in general, but did not in certain test scenarios where we did not initialize the database.  This is one solution.

rawwar committed:
add warning log when task_key>100 (#42813)

related to #41816

Adds a warning log to indicate failure if length of task_key>100.

ashb committed:
Switch `kuberenetes_tests` to run with `uv run` (#42836)

This is a no-op change right now, but as part of the provider re-org in #42505
this sets us up to be able to load the providers code in the tests

The reason this change is done separately is that changes to breeze code form
forks doesn't take effect, and this small change makes it easier to land on
main without having to re-create that large PR.

dirrao committed:
CI providers discovery update when airflow min 2.7.0 (#42767)

kaxil committed:
Make Scarf details more prominent in 2.10 release notes (#42838)

This PR moves Scarf entry in Release notes at top of 2.10.0 entries and also
adds URL/IP-address info that is covered by [Scarf's privacy policy](https://about.scarf.sh/privacy-policy).

b-per committed:
Add ability to provide proxy for dbt Cloud connection (#42737)

* Add ability to provide proxy for dbt Cloud connection

* Running pre-commit checks

* Update current tests and add new test with proxy

kaxil committed:
Update CODEOWNERS to update Kaxil's entry (#42839)

I haven't been that active coding or reviewing. Although I am now returning back to focus on Task SDK (AIP-72) so narrowing down my entries in CODEOWNERS.

glostis committed:
Remove unused ldap3 dependency (#42829)

zachliu committed:
Improve docs on adding Google secrets in AWS Secrets Manager (#42832)

The [existing documentation](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/secrets-backends/aws-secrets-manager.html#example-of-storing-google-secrets-in-aws-secrets-manager) on how to set up Google secrets in AWS Secrets Manager is out of date. It led me on a merry chase for hours.

I hereby submit this PR to update the doc. I solemnly swear that the content has been verified using DAG code similar to this

```python
gsheet = GSheetsHook(gcp_conn_id=gcp_conn_id)
values = gsheet.get_values(
    spreadsheet_id=spreadsheet_id,
    range_=f"{sheet_name}!B1:B2",
)
`

vincbeck committed:
Remove `sqlalchemy-redshift` dependency from Amazon provider (#42830)

kaxil committed:
Simplify code for recent dbt provider change (#42840)

Just simplified redundant code from https://github.com/apache/airflow/pull/42737

harjeevanmaan committed:
- Fixes #42432 (#42770)

- Added a unit test

bugraoz93 committed:
AIP-84 Migrate views /object/historical_metrics_data to Fast API (#42629)

* Include object router and migrate objects/historical_metrics_data to FastAPI

* Use pyfixture session

* Include provide_session for SQLite session thread issue

* Make method async and remove unused session from tests

* Include return type to generate proper openapi spec

* Object definition to Dashboard definition, make safe date Annotated parameter in parameters.py

* Include pydantic model for Dashboard Historical Metric Data and convert response to HistoricalMetricDataResponse

* Fix conflict and rerun pre-commit hooks

* Rename data variable to historical metrics for consistency

* Include object router and migrate objects/historical_metrics_data to FastAPI

* Object definition to Dashboard definition, make safe date Annotated parameter in parameters.py

* Drop data from method name to prevent double Data in the name of objects

* Variable name change to prevent shadow naming with method name

* Resolve conflicts and rebase again

* Resolve conflicts and rebase

tirkarthi committed:
Add support to filter by last dagrun state in UI. (#42779)

* Add support to filter by last dagrun state in UI.

* Fix lint errors.

* Fix lint errors.

* Fix PR comments over null checks and query parameter to be last_dag_run_state.

Lee-W committed:
feat(datasets): make strict_dataset_uri_validation default to True (#41814)

* feat(datasets): make strict_dataset_uri_validation default to True

* docs(assets): add uri scheme restrict instruction for AIP-60

* feat(assets): add an error logging to URI that is not AIP-60 compliant

* Better test string

---------

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

gopidesupavan committed:
Standard provider bash operator (#42252)

romsharon98 committed:
Fix mark as success when pod fails while fetching log (#42815)

* fix: always defer once more after log fetching to ensure pod completion is handled

* add tests

* removing print

---------

Co-authored-by: Jean-Eudes Peloye <peloye.je@gmail.com>

jbampton committed:
Fix spelling; `Airlfow` -> `Airflow` (#42855)

rawwar committed:
Docs: Add Template field related info for python operators (#42847)

closes #39584

This PR adds templating information related to Python operators in How-to Guide

pierrejeambrun committed:
AIP-84 Get Variable (#42834)

Handle redacted value for Variables

dstandish committed:
Move test of DagRun.update_state to better place (#42845)

Previously this lived in test_scheduler_job.py

It only really tested the behavior of DagRun.update_state.

As far as I can tell, it checks that if you null out the state on a TI of a finished dag, and then you call ``update_state``, then the DR will be set to running.

mobuchowski committed:
Revert "Remove `sqlalchemy-redshift` dependency from Amazon provider (#42830)" (#42864)

This reverts commit 3b4c73a644f40013927fb72227dfe018dc47a97f.

eladkal committed:
Prepare docs for Oct 1st adhoc wave of providers (#42862)

* Prepare docs for Oct 1st adhoc wave of providers

* Update

* update amazon

dannyl1u committed:
fix more PT004 PyDocStyle checks (#42841)

kaxil committed:
Revert Asset to Dataset for Core Extension Doc (#42867)

This PR reverts Core Extension doc change (https://airflow.apache.org/docs/apache-airflow-providers/core-extensions/index.html) from https://github.com/apache/airflow/pull/41348

Since those docs only work on Airflow stable, we can only change this after 3.0

ashb committed:
Split providers out of the main "airflow/" tree into a UV workspace project (#42505)

This is only a partial split so far. It moves all the code and tests, but
leaves the creation of `core/` to a separate PR as this is already large
enough.

In addition to the straight file rename the other changes I had to make here
are:

- Some mypy/typing fixes.

  Mypy can be fragile about what it picks up when, so maybe some of those
  changes were caused by that. But the typing changes aren't large.

- Improve typing in common.sql type stub

  Again, likely a mypy file oddity, but the types should be safe

- Removed the `check-providers-init-file-missing` check

  This isn't needed now that airflow/providers shouldn't exist at all in the
  main tree.

- Create a "dev.tests_common" package that contains helper files and common
  pytest fixtures

  Since the provider tests are no longer under tests/ they don't automatically
  share the fixtures from the parent `tests/conftest.py` so they needed
  extracted.

  Ditto for `tests.test_utils` -- they can't be easily imported in provider
  tests anymore, so they are moved to a more explicit shared location.

In future we should switch how the CI image is built to make better use of UV
caching than our own approach as that would remvoe a lot of custom code.

sonu4578 committed:
Improving validation of task retries to handle None values (#42532)

* Improving validation of task retries to handle None values

* Updated the validation check to use the "or" logical operator

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

* Added a comment for the new validation check

* Update baseoperator.py

* Update taskinstance.py

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

kaxil committed:
Fix deprecated stage names for Pre-commit (#42872)

Before
```
❯ pre-commit install
[WARNING] top-level `default_stages` uses deprecated stage names (commit, push) which will be removed in a future version.  run: `pre-commit migrate-config` to automatically fix this.
pre-commit installed at .git/hooks/pre-commit
```

pierrejeambrun committed:
Fix ui lint pre-commit interactive mode (#42854)

gopidesupavan committed:
update k8s tests urllib3 retry config status_forcelist and allowed_methods (#42871)

kaxil committed:
Bump `blacken-docs` pre-commit (#42880)

Bump `blacken-docs` pre-commit from 1.18.0 -> 1.19.0

kaxil committed:
Exclude "not-ready" providers when building docs (#42873)

This fixes the issue we saw in https://apache-airflow.slack.com/archives/C03G9H97MM2/p1728492002477169

This PR excludes the providers that aren't ready yet for building docs.

gopidesupavan committed:
Move Hooks to Standard provider (#42794)

pierrejeambrun committed:
Fix dag warning documentation (#42858)

eladkal committed:
Fix issue generation for provider after folder restructure (#42883)

omkar-foss committed:
AIP-84 Migrate the public endpoint Get DAG to FastAPI (#42848)

* Migrate the public endpoint Get DAG to FastAPI

* Use proper name for test function

rawwar committed:
AIP-84 Migrate GET Dag Run endpoint to FastAPI (#42725)

* get dag_run init

* add serializer

* Merge branch 'main' of https://github.com/apache/airflow into kalyan/AIP-84/get_dag_run

* add types

* add test

* working tests

* add note to DagRunResponse

* add note

* add test to test non Null note

* Update airflow/api_fastapi/views/public/dag_run.py

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

* Update airflow/api_fastapi/views/public/dag_run.py

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

* Merge branch 'main' of https://github.com/apache/airflow into kalyan/AIP-84/get_dag_run

* add 404 test

---------

Co-authored-by: Pierre Jeambrun <pierrejbrun@gmail.com>

Commit:
add majorosdonat as bosch user (#42890)

Co-authored-by: Majoros Donat (XC-DX/EET2-Bp) <donat.majoros2@hu.bosch.com>

vincbeck committed:
Move the session auth backend to FAB auth manager (#42878)

vincbeck committed:
Move user and roles schemas to fab provider (#42869)

dstandish committed:
Add support for run conf to backfill (#42865)

This actually does a little bit more.  It changes the backfill create endpoint to take a json payload instead of just query params.  This is just easier because we can use the backfill schema as the schema of the request body.

One thing that is maybe weird is I add a decorator to translate the request body to the kwargs in the endpoint function.  The main motivator here was for compatibility with the requires_access_dag decorator, which doesn't check request body.

pierrejeambrun committed:
Fix main (#42903)

dstandish committed:
Fix main: js/types/api-generated.ts (#42906)

gopidesupavan committed:
mark TestGKEStartKueueInsideClusterOperator tests with flaky decorator (#42916)

dstandish committed:
Remove dag.run() method (#42761)

This method uses Backfill internally. Before we can remove BackfillJobRunner, we need to remove DAG.run. But before we can remove DAG.run, we need to update some old tests that use it. So this is the first step towards removing BackflilJobRunner.

There were some very old tests that came from airflow github issue 1225. These appeared to test the scheduler but really they tested the backfill job runner. Just to be cautious, I kept most of them rather than remove (which probably would have been fine since they essentially tested code that we'll be removing). As appropriate I either changed them to run on dag.test or scheduler. The ones dealing with ignore first depends on past will have to be added back when that functionality is implemented in new backfill.

dstandish committed:
Add test for behavior for paused backfill (#42837)


-----
Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

simi committed:
Use url_from_endpoint inside HttpHook. (#42785)

kaxil committed:
Fix typo in Breeze (#42919)

A couple of minor typos due to newlines

infused-kim committed:
Docs: Add templating info to TaskFlow tutorial (#42887)

Rasnar committed:
Add possibility to override the conn type for Druid (#42793)

* Add possibility to override the conn type for Druid

Minor fix, which allows to use the schema which are specified in
theschema rather than `http` as default. In the same time it doesn't
changethe logic as any conn_type can be selected. Intuitevely it's
expectedthat anything specified in `schema` field will actually take
precedencein the building the desired url.

* Add druid endpoint connection from another PR

* Fix missing scheme in test

* Set schema to None where it's unused

Even though we don't need it directly set, by default the mock will set
it to an internal object, thus we need to override it to None.

---------

Co-authored-by: Oleg Auckenthaler <github.sitcom838@passmail.net>

rawwar committed:
fix: HttpSensorTrigger to include `method` when serializing (#42925)

pierrejeambrun committed:
AIP-84 Fix dag display name search (#42863)

* Fix dag display name search

* Fix CI

gopidesupavan committed:
Check _is_canary_run/pr condition in is_legacy_ui_api_labeled method (#42921)

* check _is_canary_run condition in is_legacy_ui_api_labeled method

* include pr check in is_legacy_ui_api_labeled

ashb committed:
Allow python 3.12 for the breeze release commands (#42936)

I'm not 100% sure why we added this check in the first place, but it doesn't
seem to be needed anymore (I've tested things locally with this removed and it
all seems to behave itself)

bbovenzi committed:
Render errors when getting a list of dags (#42897)

* Render errors when getting a list of dags

* Restore axios, prettierignore pnpm-store

* Add pnpm-store to prettier ignore

kaxil committed:
Remove the referrer from Webserver to Scarf (#42901)

This will make sure we don't receive any information about the Webserver URL sending the info like the number of plugins and such.

vlieven committed:
Split providers out of the main "airflow/" tree into a UV workspace project (#42505) (#42624)

This is only a partial split so far. It moves all the code and tests, but
leaves the creation of `core/` to a separate PR as this is already large
enough.

In addition to the straight file rename the other changes I had to make here
are:

- Some mypy/typing fixes.

  Mypy can be fragile about what it picks up when, so maybe some of those
  changes were caused by that. But the typing changes aren't large.

- Improve typing in common.sql type stub

  Again, likely a mypy file oddity, but the types should be safe

- Removed the `check-providers-init-file-missing` check

  This isn't needed now that airflow/providers shouldn't exist at all in the
  main tree.

- Create a "dev.tests_common" package that contains helper files and common
  pytest fixtures

  Since the provider tests are no longer under tests/ they don't automatically
  share the fixtures from the parent `tests/conftest.py` so they needed
  extracted.

  Ditto for `tests.test_utils` -- they can't be easily imported in provider
  tests anymore, so they are moved to a more explicit shared location.

In future we should switch how the CI image is built to make better use of UV
caching than our own approach as that would remvoe a lot of custom code.

Co-authored-by: Ash Berlin-Taylor <ash@apache.org>
Co-authored-by: Ryan Hatter <25823361+RNHTTR@users.noreply.github.com>

gopidesupavan committed:
increase backoff_factor and add try/catch in k8s tests (#42940)

potiuk committed:
Consistent python version checks and troubleshooting (#42944)

Follow up after #42766 and #42936.

* We do not have to check for minimum Python version for Python 3.9
  any more (as we do not support 3.8 any more)
* While Python 3.12 is not yet fully supported by Apache Beam, we
  should still not allow it for releasing providers, but all other
  commands should support Python 3.12
* When you had pre-commit installed with Python 3.8 before, various
  errors might appear when running pre-commit. Troubleshooting
  was added quoting the errors and explaining what to do.

Commit:
Update README.rst (#42941)

grammar update for clarity 😄 👍

dstandish committed:
Add a "backfill create" command (#42922)

I create a new command group "backfill" for management of backfills. The first action is "create" which creates a backfill. Some others may follow such as pause / cancel.

sfc-gh-jdu committed:
Fix SNOWFLAKE_CONN_ID and DAG_ID in Snowpark system tests (#42952)

kaxil committed:
Fix broken links in Release Management docs (#42958)

These links were pointing to the wrong location. This PR fixes it

gopidesupavan committed:
Restrict looker-sdk version 24.18.0 and microsoft-kiota-http 1.3.4 (#42954)

* restrict looker version 24.18.0

* update microsoft provider deps with microsoft-kiota-http

* update gh issue ids to provider yaml files

* ignore type in applfowhook

* ignore type in applfowhook

* ignore type in applfowhook

gopidesupavan committed:
upgrade trove-classifiers (#42950)

Aakcht committed:
Chart: fix VCT for scheduler in local and persistent mode (#42946)

peloyeje committed:
fix: use instance base_container_name to fetch logs on trigger_reentry (#42960)

kgw7401 committed:
fix: 03_contributors_quick_start_docs (#42927)

* Change code-block position

kaxil committed:
Add skeleton project for task-sdk (#42904)

closes https://github.com/apache/airflow/issues/42856

This PR adds a skeleton project for [AIP-72 Task SDK](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-72+Task+Execution+Interface+aka+Task+SDK)

It creates this new project, integrates them to breeze & runs it on CI.

The code & tests are just placeholders at this point, we will start adding things needs for DAG Authors in separate PRs.

kaxil committed:
Upgrade Trove classifier to `2024.10.12` (#42961)

Test was failing: https://github.com/apache/airflow/actions/runs/11306055404/job/31446236535

```
pre-commit hook(s) made changes.
If you are seeing this message in CI, reproduce locally with: `pre-commit run --all-files`.
To run `pre-commit` as part of git workflow, use `pre-commit install`.
All changes made by hooks:
diff --git a/pyproject.toml b/pyproject.toml
index 844e1b6..5e9f587 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -30,7 +30,7 @@ requires = [
     "pluggy==1.5.0",
     "smmap==5.0.1",
     "tomli==2.0.2; python_version < '3.11'",
-    "trove-classifiers==2024.10.11",
+    "trove-classifiers==2024.10.12",
 ]
 build-backend = "hatchling.build"
```

kaxil committed:
Add missed brackets for our dev script for Spell checks (#42965)

A nit PR while I was looking at our doc builds

dabla committed:
FIX: Don't raise a warning in ExecutorSafeguard when execute is called from an extended operator (#42849)

* refactor: Don't raise a warning when execute is called from an extended operator, as this should always be allowed.

* refactored: Fixed import of test_utils in test_dag_run

---------

Co-authored-by: David Blain <david.blain@infrabel.be>

potiuk committed:
Upgrade Helm Chart dependencies to latest released (#42816)

jason810496 committed:
Fix PythonOperator DAG error when DAG has hyphen in name (#42902)

potiuk committed:
fIX handling removal of dependencies (#42967)

There was a problem with CI image builds when cache has been
disabled - they still used "pre-cached" main version (unlike
PROD builds). This PR synchronizes the behaviour between CI
and PROD builds.

For removing dependencies, you need to set both:

* `disable docker cache` label
* incerase DEPENDENCIES_EPOCH_NUMBER in `Dockerfile.ci`

Comments and documentation in both places has been updated to
reflect it.

Since documentation for labels has been updated,
Part of this PR is to improve the description of possible
labels that could be used during the build.
The description grew from a small number of labels to a
"wall of text" that was difficult to read. This PR reformats it in the
form of table that makes it far easier to see actions that the
maintainer can do and what labels should be set for each of them.

gopidesupavan committed:
mark test_task_workflow_trigger_success as flaky (#42972)

AryanK1511 committed:
Create a User Settings button with light/dark mode toggle as a menu item (#42964)

rawwar committed:
AIP-84 Migrate delete Dag Run endpoint to FastAPI (#42910)

* delete dag_run init

* delete dag_run init

* add delete dag run endpoint

* add tests for delete dag run

michaeljs-c committed:
Make datStats endpoint dag_ids parameter optional (#42955)

* Make datStats dag_id parameter optional with pagination

* move pagination out of sql query

* tidy

---------

Co-authored-by: Michael Smith-Chandler <mjsmithandler@gmail.com>

gopidesupavan committed:
trove classifier upgrade (#42979)

vincbeck committed:
Change directory used by simple auth manager to store generated passwords (#42860)

* Change directory used by simple auth manager to store generated passwords

* Update airflow/auth/managers/simple/simple_auth_manager.py

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

---------

Co-authored-by: Jens Scheffler <95105677+jscheffl@users.noreply.github.com>

gopidesupavan committed:
Disable flaky mssql based integration tests (#42811)

* quarantined flaky mssql integration tests

* disable mssql from ci integration checks

dstandish committed:
Remove BackfillJobRunner class (#42943)

olegkachur-e committed:
Add early job_id xcom_push for google provider Beam Pipeline operators (#42982)

- To let GCP Beam Sensor operators 'sense' the pipeline changes,
by having dataflow job_id been xcom_push as soon as it available.

Related issue: https://github.com/apache/airflow/issues/30007.

Co-authored-by: Oleg Kachur <kachur@google.com>

dstandish committed:
Add logic to mark backfills as complete (#42683)

Periodically check for backfills that should be marked as complete.
